src/modules/esl-popup/core/esl-popup.ts

```diff
@@ -91,32 +97,36 @@ export class ESLPopup extends ESLToggleable {
   @prop() public closeOnOutsideAction = true;
 
   @ready
-  public connectedCallback() {
+  public connectedCallback(): void {
     super.connectedCallback();
     this.$arrow = this.querySelector('span.esl-popup-arrow');
   }
 
-  /** Is position along horizontal axis? */
-  protected get _isPositioningAlongHorizontal() {
-    return ['left', 'right'].includes(this.position);
+  /** Checks that the position along the horizontal axis */
+  @memoize()
+  protected get _isMajorAxisHorizontal(): boolean {
+    return isMajorAxisHorizontal(this.position);
   }
 
-  /** Is position along vertical axis? */
-  protected get _isPositioningAlongVertical() {
-    return ['top', 'bottom'].includes(this.position);
+  /** Checks that the position along the vertical axis */
+  @memoize()
+  protected get _isMajorAxisVertical(): boolean {
+    return !isMajorAxisHorizontal(this.position);
   }
```

Comment:
Cosmetic: it looks like the memoize will be much heavier here then exact check inside the `isMajorAxisHorizontal`

Reply:
I was of the same opinion at first.
But I wondered to check and understand with the help of benchmarks. I don't seem to have the same understanding now. 
We can discuss it.